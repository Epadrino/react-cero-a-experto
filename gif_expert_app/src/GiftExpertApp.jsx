import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GiftExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch', 'dragon Ball']);

	const onAddCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;
		setCategories([newCategory, ...categories]);
		// setCategories = [...categories, "Valoran"];
	};
	return (
		<>
			{/* titulo */}
			<h1>GiftExpertApp</h1>

			{/* imput */}
			<AddCategory
				// setCategories={setCategories}
				onNewCategory={(value) => onAddCategory(value)}
			/>

			{/* listado de Gif */}

			<ol>
				{categories.map((category) => {
					return <li key={category}>{category}</li>;
				})}
			</ol>
			{/* Gif Item */}
		</>
	);
};
