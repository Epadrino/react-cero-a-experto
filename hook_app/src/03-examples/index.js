export * from './CardPokemons';
export * from './Loading';
export * from './MultiplesCustomHooks';
