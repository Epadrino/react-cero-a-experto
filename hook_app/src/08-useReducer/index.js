export * from './TodoApp';
export * from './TodoList';
export * from './TodoAdd';
export * from './todoReducer';
export * from './TodoItem';
