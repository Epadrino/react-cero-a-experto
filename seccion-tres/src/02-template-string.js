const nombre = "Eduardo";
const apellido = "Padrino";

// concatenado normal
const nombreCompleto = nombre + " " + apellido;

console.log(nombreCompleto);

// concatenado moderno
const nombreCompletoConString = `
${nombre} 
${apellido}
${1 + 1}`;

console.log(nombreCompletoConString);

const getSaludo = (nombre) => {
  return "hola " + nombre;
};

// impresion en consola con string
console.log(`Este es un texto: ${getSaludo(nombre)}`);
