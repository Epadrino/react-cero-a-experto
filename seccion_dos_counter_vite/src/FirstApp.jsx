import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle} </p>
      <p>{subTitle} </p>
      <p>explicacion del nodo padre </p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
  title: "agrega un Titulo",
  subTitle: "Agrega un Subtitulo",
};
