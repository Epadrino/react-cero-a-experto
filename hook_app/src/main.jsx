import React from 'react';
import ReactDOM from 'react-dom/client';
// import { BrowserRouter } from 'react-router-dom';
// import { MainApp } from './09-useContext/MainApp';
// import { SimpleForm } from './02-useEffect/SimpleForm';
// import { CounterApp } from './01-useState/CounterApp';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { HooksApp } from './HooksApp';
import './08-useReducer/intro-reducer';
import './index.css';

// ReactDOM.createRoot(document.getElementById('root')).render(
// 	<BrowserRouter>
// 		<React.StrictMode>
// 			<MainApp />
// 		</React.StrictMode>
// 	</BrowserRouter>
// );
