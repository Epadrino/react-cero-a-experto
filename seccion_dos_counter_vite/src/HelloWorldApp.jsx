import { FirstApp } from "./FirstApp";

export const HelloWorldApp = () => {
  return <h1>HelloWorldApp </h1>;
};
