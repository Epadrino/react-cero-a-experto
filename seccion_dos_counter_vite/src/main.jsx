import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";
import { HelloWorldApp } from "./HelloWorldApp";
import "./styleds.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <HelloWorldApp />
    <FirstApp title="Esto es una Props" />
  </React.StrictMode>
);
