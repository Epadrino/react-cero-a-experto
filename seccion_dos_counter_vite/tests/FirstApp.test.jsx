import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp/>>", () => {
  test("should hacer match con el snapshot", () => {
    const title = "hola, Soy Goku";
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test("should de mostrar el titulo en un h1", () => {
    const title = "hola, Soy Goku";
    const { container, getByText } = render(<FirstApp title={title} />);
    expect(getByText(title)).toBeTruthy();
    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
  });
});
