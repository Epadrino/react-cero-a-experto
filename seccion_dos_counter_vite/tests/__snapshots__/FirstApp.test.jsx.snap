// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/>> should hacer match con el snapshot 1`] = `
<div>
  <h1>
    hola, Soy Goku
  </h1>
  <p>
    explicacion del nodo padre 
  </p>
</div>
`;
